### PR TITLE
refactor: displays CAN obligate_by as MM/DD/YY

### PR DESCRIPTION
**Pull request by fpigeonjr**
## Issue

#346 

## How to test

1. Go to list of cans
2. see obligate_by as MM/DD/YY

## Screenshots

image

## Definition of Done Checklist
- [X] OESA: Code refactored for clarity
- [X] OESA: Dependency rules followed
- [X] Automated unit tests updated and passed
- [X] Automated integration tests updated and passed
- [X] Automated quality tests updated and passed
- [X] Automated load tests updated and passed
- [X] Automated a11y tests updated and passed
- [X] Automated security tests updated and passed
- [X] 90%+ Code coverage achieved
- [-] Form validations updated

### CODE REVIEW

File: backend/openapi.yml [hunk 1]
```diff
@@ -2371,8 +2373,14 @@ components:
           type: array
           items:
             $ref: "#/components/schemas/FundingReceived"
+        id:
+          type: integer
+        nick_name:
+          type: string
         number:
           type: string
+        obligate_by:
```
Comment (fpigeonjr) [inline backend/openapi.yml:2382]: added `obligate_by` and sorted alphebetically

File: frontend/src/components/CANs/CANTable/CANTable.jsx [hunk 2]
```diff
@@ -56,67 +57,6 @@ const CANTable = ({ cans }) => {
     );
 };
 
-const TableHead = () => {
```
Comment (fpigeonjr) [inline frontend/src/components/CANs/CANTable/CANTable.jsx:59]: moved to its own component

File: frontend/src/components/CANs/CANTypes.d.ts [hunk 3]
```diff
@@ -17,13 +15,15 @@ export type CAN = {
     id: number;
     nick_name?: string;
     number: string;
-    obligate_by?: Date;
+    obligate_by?: number;
```
Comment (fpigeonjr) [inline frontend/src/components/CANs/CANTypes.d.ts:18]: @rajohnson90 didn't think this was a Date

**Review by maiyerlee — APPROVED** [review]

File: frontend/src/components/CANs/CANTypes.d.ts (hunk 3 above)
Comment (rajohnson90) [reply frontend/src/components/CANs/CANTypes.d.ts:18]: It isn't a date. Already covered this on collab but just writing it down so it's documented.

**Review by rajohnson90 — APPROVED** [review]

**Review by johndeange — CHANGES_REQUESTED** [review]
I think this PR has the off by one error that we discussed the other day, e.g. in the example a 1 year CAN with an appropriation year of 2023 (10/1/2022) should be valid until 9/30/2023 (according to our current business rules).  In this case, when the backend returns 2024, the 2024 fiscal year begins 10/1/2023 and so the obligate by should be 9/30/2023.

**Comment by fpigeonjr** [comment]
> I think this PR has the off by one error that we discussed the other day, e.g. in the example a 1 year CAN with an appropriation year of 2023 (10/1/2022) should be valid until 9/30/2023 (according to our current business rules). In this case, when the backend returns 2024, the 2024 fiscal year begins 10/1/2023 and so the obligate by should be 9/30/2023.

I was scared of this, thanks John for calling this out.

**Comment by fpigeonjr** [comment]
image

**Pull request merged**